### PR TITLE
added spm, macOS testing, and pod lint to travis

### PR DESCRIPTION
**Pull request by dfirsht**
DCO1.1 Signed-off-by: Daniel Firsht <dfirsht@us.ibm.com>

### CODE REVIEW

**Comment by boopt2** [comment]
Hi dfirsht,

Thanks for submitting this pull request!

Please add a comment with a DCO1.1 Signed-off-by statement in order to allow us to process your pull request.

dco-bot

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,11 @@ before_script:
 script: 
   - xcodebuild -project CSyncSDK.xcodeproj -scheme CSyncSDK-iOS -destination platform='iOS Simulator',name='iPhone 6',OS='10.3' clean test
   - xcodebuild -project CSyncSDK.xcodeproj -scheme CSyncSDK-iOS -destination platform='iOS Simulator',name='iPhone 6',OS='9.3' clean test
+  - xcodebuild -project CSyncSDK.xcodeproj -scheme CSyncSDK-OSX -destination platform='macOS' clean test
+  - swift -version
```
Comment (boopt2) [inline .travis.yml:15]: Why are you running swift -version?

Comment (dfirsht) [reply .travis.yml:15]: just so we can look in the logs and see what version is running, could be useful for debugging later

**Review by boopt2 — APPROVED** [review]
Looks good

**Pull request merged**